rust/tools/authorization-logic/src/compilation_top_level.rs

```diff
@@ -0,0 +1,27 @@
+use std::fs;
+use crate::ast::*;
+use crate::parsing::astconstructionvisitor;
+use crate::souffle::souffle_interface;
+use crate::signing::export_assertions;
+use crate::signing::import_assertions;
+
+// This has the top level function to compile to souffle. This is a separate file and module from 
+// souffle-interface because part of the compilation step involces importing and exporting claims. 
+// The importing and exporting of claims happens at the AST level rather than an IR and works 
+// independently of the fact that the particular choice of solver is datalog. This separation might 
+// make it easier to switch to a different solver later. This is also separate from main because
+// "compile" is also used for tests
+
+fn source_file_to_ast(filename: &String, in_dir: &String) -> AstProgram {
```

Comment:
nit: It is usually preferred to use &str in Rust rather than &String (see e.g. https://stackoverflow.com/questions/40006219/why-is-it-discouraged-to-accept-a-reference-to-a-string-string-vec-vec-o)

Reply:
Suggestion applied broadly.